x/did/types/did.go

```diff
@@ -560,6 +561,25 @@ func (didDoc DidDocument) HasRelationship(
 	return false
 }
 
+// HasPublicKey validates if a public key is contained in a DidDocument
+func (didDoc DidDocument) HasPublicKey(pubkey cryptotypes.PubKey) bool {
```

Comment:
there are more options the PublicKeyHex and BlockchainAccountId (see toAddress) in the latest verison (REMINDER)